light-clients/common/src/lib.rs

```diff
@@ -194,8 +194,9 @@ pub fn decode_timestamp_extrinsic(ext: &Vec<u8>) -> Result<u64, anyhow::Error> {
 	// https://github.com/paritytech/substrate/blob/d602397a0bbb24b5d627795b797259a44a5e29e9/primitives/trie/src/lib.rs#L99-L101
 	// Decoding from the [2..] because the timestamp inmherent has two extra bytes before the call
 	// that represents the call length and the extrinsic version.
-	let (_, _, timestamp): (u8, u8, Compact<u64>) = codec::Decode::decode(&mut &ext[2..])
-		.map_err(|err| anyhow!("Failed to decode extrinsic: {err}"))?;
+	let (_, _, timestamp): (u8, u8, Compact<u64>) =
+		codec::Decode::decode(&mut &ext[ext.len() - 7..])
```

Comment:
just checked that it is fine to use &ext[5..]
and need to decode directly into Compact<u64> and not (u8, u8, Compact<u64>)

so it supposed to be like this

```
let ret: Compact<u64> = codec::Decode::decode(&mut &ext[5..])
		.map_err(|err| anyhow!("Failed to decode extrinsic: {err}"))?;
```